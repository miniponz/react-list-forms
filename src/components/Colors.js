
import React from 'react';
import PropTypes from 'prop-types';
import Color from './Color';

function Colors({ colors }) {

  const colorList = colors.map(({ name, hex, rgb }) => {
    <li>
      <Color key={`${name}, ${hex}, ${rgb}`} name={name} hex={hex} rgb={rgb} />
    </li>;
  });

  return (
    <ul>
      {colorList}
    </ul>
  );
}

Colors.propTypes = {
  colors: PropTypes.array.isRequired
};

export default Colors;


