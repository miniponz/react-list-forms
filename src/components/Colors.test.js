import React from 'react';
import { shallow } from 'enzyme';
import Colors from './Colors';

describe('colors list', () => {
  it('renders color list', () => {
    const wrapper = shallow(<Colors colors={[]}/>);
    expect(wrapper).toMatchSnapshot();
  });   
});
