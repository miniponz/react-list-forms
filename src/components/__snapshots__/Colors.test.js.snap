// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colors list renders color list 1`] = `<ul />`;
